### DISCUSSION TITLE
2 questions pour intégration de Tac dans un plugin Wordpress Tout-en-un

### DISCUSSION THREAD

**Discussion by glb-blg** [AmauriC/tarteaucitron.js#1143] [General]
Bonjour Amauri,

Merci pour tout ton travail sur ce tarteaucitron !!

Je travaille sur un plugin Wordpress RGPD tout-en-un : configuration en admin (pour 15/20 services) + obfuscation des balises à la volée (pour 4/5 éditeurs de contenus et pour 15/20 plugins) etc...

Dans ce cadre, je me permets de "hooker" tarteaucitron dans certains cas.

Exemple pour la gestion du plugin monsterInsights, je capture/supprime tout le javascript relatif à googletagmanager que le plugin tente d'écrire dans le HTML, je l'envoie côté navigateur et si l'utilisateur a accepté le service, je rétablis ce javascript. Dans ce cas, j'ai besoin de "supprimer" la fonction tarteaucitron.services.gtag.js() pour ne pas qu'elle rajoute l'appel du script https://www.googletagmanager.com. 
	J'ai trouvé la solution : **window.tarteaucitronMagic.push('\_gtag\_')**;
	Dans ce cas, la fonction youtubeService.js() n'est pas appelée.

**Mais est-ce une solution viable et pérenne (car je ne vois pas trop de documentation sur cette variable "tarteaucitronMagic") ?**

De plus, en plus de pouvoir paramétrer les services dans l'admin du plugin, j'aimerais aussi laisser la possibilité de pointer sur https://tarteaucitron.io/load.js. Mais ça risque d'être compliqué avec tous les hooks que je développe 😓. Je ne vais pas tarder à me pencher dessus sérieusement mais je profite de ce post pour te demander : **n'y aurait-il pas par hasard un accès similaire mais pour recevoir la config en JSON (par exemple) ?**

Merci d'avance !

Christophe

**Comment by AmauriC** [comment]
Bonjour Christophe,

J'accueil avec plaisir tous les plugins tiers mais il y a 2 problèmes ici:

- les fonctions magic de tarteaucitron concernent la v pro de tarteaucitron et ne sont donc pas documentés et je ne recommande pas l'usage, ça risque de créer des conflits
- idem pour le chargement du tag depuis mes serveurs (tarteaucitron.io), c'est réservés aux clients qui me paie, je ne fourni pas le cdn gratuitement ; )

Il y a toute la config de tous les services dispo ici: https://tarteaucitron.io/json.php
C'est sûrement ce que tu cherches ;)

**Reply by mably** [reply]
Bonjour @AmauriC, le lien ci-dessus ne semble plus fonctionnel.

Serait-il possible de nous communiquer la nouvelle URL stp?

En te remerciant par avance.

**Reply by AmauriC** [reply]
Salut !
C'est de nouveau dispo avec la mise à jour à venir pour ajouter `data-` devant tous les attributs.
A utiliser à partir de la v1.20 qui arrive demain si tout est ok.

**Reply by mably** [reply]
Super, merci !
